### PR TITLE
allow expression in IN builder for columns

### PR DESCRIPTION
**Pull request by ntesic**
| Q             | A
| ------------- | ---
| Is bugfix?    | ✔️
| New feature?  | ❌
| Breaks BC?    | ❌
| Fixed issues  | #18821

### CODE REVIEW

**Review by samdark — APPROVED** [review]
Looks good. Would you please add a line for CHANGELOG?

File: framework/CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,4 @@
+Enh #18821: Allow `yii\db\ExpressionInterface` as column in `yii\db\conditions\InBuilder`
```
Comment (bizley) [inline framework/CHANGELOG.md:1]: I have once fixed the changelog you made previously like that - please put it in the right place.

**Comment by machour** [comment]
@ntesic I'm genuinely failing to see a use case for expressions in IN. Can you provide a real-world case where you had to use this?

**Comment by ntesic** [comment]
@machour I can't say any real-world case, just was inspired with issue #18821

**Comment by machour** [comment]
Thanks for getting back to me, and thank you for the code contribution!

I feel like there isn't really a need for this feature (even reading through the original issue), and we're adding a bunch of ifs for something unlikely to be used, but if the rest of the @yiisoft/reviewers deems it useful why not.

**Review by Arhell — APPROVED** [review]

**Comment by samdark** [comment]
I also think that we should not add such enhancements without demand.

@ntesic thanks for contributing but I think this PR and issue are to be closed this time.

**Comment by samdark** [comment]
Oh, wait. Re-read original issue. It is valid.

**Review by samdark — APPROVED** [review]

File: framework/db/conditions/InConditionBuilder.php [hunk 2]
```diff
@@ -66,6 +66,10 @@ public function build(ExpressionInterface $expression, array &$params = [])
             $column = $column->current();
         }
 
+        if ($column instanceof ExpressionInterface) {
```
[the remark this answers is not in the document]
Comment (samdark) [reply framework/db/conditions/InConditionBuilder.php:69]: https://github.com/yiisoft/yii2/pull/19217

**Pull request merged**